### PR TITLE
TL Localization - Chapter 7, Submodules (1,186 words)

### PR DESCRIPTION
**Pull request by heywens**
Translation for submodules.asc - part 2
Word count not including commands displayed on console.
Kindly check this contribution.

Thanks.

### CODE REVIEW

File: book/07-git-tools/sections/submodules.asc [hunk 1]
```diff
@@ -77,80 +77,80 @@ Maaari mong sapawan ang halaga na ito nang lokal gamit ang `git config submodule
 Kung naaangkop, maaaring makatulong ang isang kaugnay na URL.
 =====
 
-The other listing in the `git status` output is the project folder entry.
-If you run `git diff` on that, you see something interesting:
+Ang ibang paglilista sa `git status` na output ay ang entry ng folder ng proyekto.
+Kung papatakbuhin mo ang `git diff` doon, nakikita mo ang isang kawili-wiling bagay:
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:81]: kawili-wiling to kawili-wili na

File: book/07-git-tools/sections/submodules.asc [hunk 2]
```diff
@@ -77,80 +77,80 @@ Maaari mong sapawan ang halaga na ito nang lokal gamit ang `git config submodule
...
+indeks 0000000..c3f01dc
 --- /dev/null
 +++ b/DbConnector
 @@ -0,0 +1 @@
 +Subproject commit c3f01dc8862123d317dd46284b05b6892c7b29bc
 ----
 
-Although `DbConnector` is a subdirectory in your working directory, Git sees it as a submodule and doesn’t track its contents when you’re not in that directory.
-Instead, Git sees it as a particular commit from that repository.
+Kahit na ang `DbConnector` ay isang subdirectory ng iyong tinatrabahuang direktoryo, nakikita ito ng Git bilang isang submodule at hindi nagsusubaybay sa mga nilalaman nito kapag wala ka sa direktoryo iyon.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:95]: subdirectory  to subdirektoryo

Comment (mind-blast) [reply book/07-git-tools/sections/submodules.asc:95]: submodule  to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 3]
```diff
@@ -77,80 +77,80 @@ Maaari mong sapawan ang halaga na ito nang lokal gamit ang `git config submodule
...
+[master fb9093c] nagdagdag ng DbConnector na module
+ 2 mga file ang nabago, 4 pagsisingit(+)
+ lumikha ng mode na 100644 .gitmodules
+ lumikha ng mode na 160000 DbConnector
 ----
 
-Notice the `160000` mode for the `DbConnector` entry.
-That is a special mode in Git that basically means you’re recording a commit as a directory entry rather than a subdirectory or a file.
+Pansinin ang mode na `160000` para sa entry ng `DbConnector`.
+Iyon ay isang espesyal na mode sa Git na nangangahulugan na ikaw ay nagtatala ng isang commit bilang isang entry ng direktoryo sa halip na isang subdirectory o isang file.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:127]: subdirectory  to subdirektoryo

File: book/07-git-tools/sections/submodules.asc [hunk 4]
```diff
@@ -77,80 +77,80 @@ Maaari mong sapawan ang halaga na ito nang lokal gamit ang `git config submodule
...
+Panghuli, i-push ang mga pagbabagong ito:
 
 [source,console]
 ----
 $ git push origin master
 ----
 
 [[_cloning_submodules]]
-==== Cloning a Project with Submodules
+==== Pagkokopya ng isang Proyekto na may mga Submodule
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:137]: Submodule to Submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 5]
```diff
@@ -77,80 +77,80 @@ Maaari mong sapawan ang halaga na ito nang lokal gamit ang `git config submodule
...
 
-Here we’ll clone a project with a submodule in it.
-When you clone such a project, by default you get the directories that contain submodules, but none of the files within them yet:
+Dito, ating ikokopya ang isang proyekto na may isang submodule sa loob nito.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:139]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 6]
```diff
@@ -77,80 +77,80 @@ Maaari mong sapawan ang halaga na ito nang lokal gamit ang `git config submodule
...
+Kapag ikaw ay nagkokopya tulad ng proyektong ito, bilang default, nakakakuha ka ng mga direktoryo na naglalaman ng mga submodule, ngunit wala pa sa mga file na ito ang nasa loob nito:
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:140]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 7]
```diff
@@ -165,169 +165,169 @@ $ ls
...
+remote: Pagbibilang ng mga bagay: 11, tapos na.
+remote: Pagko-compress ng mga bagay: 100% (10/10), tapos na.
+remote: Kabuuan 11 (delta 0), muling nagamit 11 (delta 0)
+Pag-aalis ng laman ng mga bagay: 100% (11/11), tapos na.
+Pagsusuri ng pagkakakonekta... tapos na.
+Submodule path 'DbConnector': sinuri ang 'c3f01dc8862123d317dd46284b05b6892c7b29bc'
 ----
 
-Now your `DbConnector` subdirectory is at the exact state it was in when you committed earlier.
+Ngayon na ang iyong subdirectory na `DbConnector` ay nasa eksaktong estado kung nasaan ito kumakailan noong na-commit mo kanina.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:185]: subdirectory to subdirektoryo

File: book/07-git-tools/sections/submodules.asc [hunk 8]
```diff
@@ -165,169 +165,169 @@ $ ls
...
 
-There is another way to do this which is a little simpler, however.
-If you pass `--recurse-submodules` to the `git clone` command, it will automatically initialize and update each submodule in the repository.
+May ibang paraan upang gawin ito kung saan ay mas simple, gayunman.
+Kung ipapasa mo ang `--recurse-submodules` sa uto na `git clone`, ito ay awtomating magsisimula at mag-update ng bawat submodule sa repositoryo.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:188]: awtomating to awtomatikong

Comment (mind-blast) [reply book/07-git-tools/sections/submodules.asc:188]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 9]
```diff
@@ -165,169 +165,169 @@ $ ls
...
+remote: Pagbibilang ng mga bagay: 11, tapos na.
+remote: Pagko-compress ng mga bagay: 100% (10/10), tapos na.
+remote: Kabuuan 11 (delta 0), muling nagamit 11 (delta 0)
+Pag-aalis ng laman ng mga bagay: 100% (11/11), tapos na.
+Pagsusuri ng pagkakakonekta... tapos na.
+Submodule path 'DbConnector': sinuri ang 'c3f01dc8862123d317dd46284b05b6892c7b29bc'
 ----
 
-==== Working on a Project with Submodules
+==== Pagtatrabaho sa isang Proyekto na may mga Submodule
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:209]: Submodule to Submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 10]
```diff
@@ -165,169 +165,169 @@ $ ls
...
 
-Now we have a copy of a project with submodules in it and will collaborate with our teammates on both the main project and the submodule project.
+Ngayon na mayroon tayong isang kopya ng isang proyekto na may mga submodule sa loob nito at makikupagtulungan sa ating mga kasamahan sa koponan sa kapwa pangunahing proyekto at submodule na proyekto.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:211]: submodule  to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 11]
```diff
@@ -165,169 +165,169 @@ $ ls
...
 
-===== Pulling in Upstream Changes
+===== Pag-pull sa mga Pagbabago ng Upstream
 
-The simplest model of using submodules in a project would be if you were simply consuming a subproject and wanted to get updates from it from time to time but were not actually modifying anything in your checkout.
-Let's walk through a simple example there.
+Ang pinakasimpleng model ng paggamit ng mga submodule sa isang proyekto ay kung ikaw ay nag-uubos lamang ng isang subproject at nais na makakuha ng mga pagbabago mula dito sa pana-panahon ngunit hindi akwtal na nagbabago ng anuman sa iyong pagsusuri.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc]: model to modelo

Comment (mind-blast) [reply book/07-git-tools/sections/submodules.asc]: submodule to submodyul

Comment (mind-blast) [reply book/07-git-tools/sections/submodules.asc]: subproject to subproyekto

File: book/07-git-tools/sections/submodules.asc [hunk 12]
```diff
@@ -165,169 +165,169 @@ $ ls
...
+Talakayin natin ang isang simpleng halimbawa doon.
 
-If you want to check for new work in a submodule, you can go into the directory and run `git fetch` and `git merge` the upstream branch to update the local code.
+Kung gusto mong magsuri para sa bagong trabaho sa isang submodule, maaari kang pumunta sa direktoryo at patakbuhin ang `git fetch` at `git merge` ang upstream na branch upang ma-update ang lokal na code.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 13]
```diff
@@ -165,169 +165,169 @@ $ ls
...
 Fast-forward
  scripts/connect.sh | 1 +
  src/db.c           | 1 +
- 2 files changed, 2 insertions(+)
+ 2 na mga file na nabago, 2 pagsisingit(+)
 ----
 
-Now if you go back into the main project and run `git diff --submodule` you can see that the submodule was updated and get a list of commits that were added to it.
-If you don't want to type `--submodule` every time you run `git diff`, you can set it as the default format by setting the `diff.submodule` config value to ``log''.
+Kung gusto mo ngayong bumalik sa pangunahing proyekto at patakbuhin ang `git diff --submodule`, maaari mong makita na ang submodule ay na-update at makakuha ng isang listahan ng mga commit na naidagdag dito.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:233]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 14]
```diff
@@ -165,169 +165,169 @@ $ ls
...
+  > mas mahusay na kalakaran ng db
+  > mas mahusay na kalakaran ng koneksyon
 ----
 
-If you commit at this point then you will lock the submodule into having the new code when other people update.
+Kung nag-commit ka sa puntong ito pagkatapos ay ila-lock mo ang submodule sa pagkakaroon ng bagong code kapag nag-update ang iba pang mga tao.
 
-There is an easier way to do this as well, if you prefer to not manually fetch and merge in the subdirectory.
-If you run `git submodule update --remote`, Git will go into your submodules and fetch and update for you.
+Mayroong din isang mas madaling paraan upang gawin ito, kung naisin mo na hindi manu-manong kunin at i-merge sa subdirectory.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:247]: subdirectory to subdirektoryo

File: book/07-git-tools/sections/submodules.asc [hunk 15]
```diff
@@ -165,169 +165,169 @@ $ ls
...
+Kung patakbuhin mo ang `git submodule update --remote`, pupuntahan ng Git ang iyong mga submodule at kunin at i-update para sa iyo.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:248]: submodule  to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 16]
```diff
@@ -165,169 +165,169 @@ $ ls
...
    3f19983..d0354fc  master     -> origin/master
-Submodule path 'DbConnector': checked out 'd0354fc054692d3906c85c3af05ddce39a1c0644'
+Submodule path 'DbConnector': sinuri ang 'd0354fc054692d3906c85c3af05ddce39a1c0644'
 ----
 
-This command will by default assume that you want to update the checkout to the `master` branch of the submodule repository.
-You can, however, set this to something different if you want.
-For example, if you want to have the DbConnector submodule track that repository's ``stable'' branch, you can set it in either your `.gitmodules` file (so everyone else also tracks it), or just in your local `.git/config` file.
-Let's set it in the `.gitmodules` file:
+Ang utos na ito ay ipagpalagay bilang defualt na nais mong i-update ang paglabas sa branch ng `master` ng repositoryo ng submodule.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:262]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 17]
```diff
@@ -165,169 +165,169 @@ $ ls
...
+Kaya mo, gayunpaman, itakda ito sa ibang bagay kung gusto mo.
+Halimbawa, kung gusto mo na subaybayan ng DbConnector na submodule ang ``stable'' na branch ng repostiryong iyon, maaari mong itakda ito sa alinman sa iyong file na `.gitmodules` (upang subaybayan din ng iba), o sa iyong lokal na `.git/config` na file lamang.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:264]: repostiryong  to repositoryong

Comment (mind-blast) [reply book/07-git-tools/sections/submodules.asc:264]: submodule  to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 18]
```diff
@@ -165,169 +165,169 @@ $ ls
...
    27cf5d3..c87d55d  stable -> origin/stable
-Submodule path 'DbConnector': checked out 'c87d55d4c6d4b05ee34fbc8cb6f7bf4585ae6687'
+Submodule path 'DbConnector': sinusuri ang 'c87d55d4c6d4b05ee34fbc8cb6f7bf4585ae6687'
 ----
 
-If you leave off the `-f .gitmodules` it will only make the change for you, but it probably makes more sense to track that information with the repository so everyone else does as well.
+Kung iiwanan mo ang `-f .gitmodules`, ito ay gagawa lamang ng pagbabago para sa iyo, ngunit malamang na ito ay mas makatutulong upang masubaybayan ang impormasyong iyon sa repositoryo upang ang iba naman ay makagagawa rin.
 
-When we run `git status` at this point, Git will show us that we have ``new commits'' on the submodule.
+Kapag patakbuhin natin ang `git status` sa puntong ito, ipapakita ng Git sa atin na mayroon tayong ``bagong mga commit'' sa submodule.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:283]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 19]
```diff
@@ -165,169 +165,169 @@ $ ls
...
-  modified:   DbConnector (new commits)
+  nabago:   .gitmodules
+  nabago:   DbConnector (mga bagong commit)
 
-no changes added to commit (use "git add" and/or "git commit -a")
+walang pagbabago naidagdag sa commit (gumamit ng "git add" at/o "git commit -a")
 ----
 
-If you set the configuration setting `status.submodulesummary`, Git will also show you a short summary of changes to your submodules:
+Kung nagtatakda ka ang setting ng pagsasaayos na `status.submodulesummary`, ipapakita din ng Git sa iyo ang isang maikling buod ng mga pagbabago sa iyong mga submodule:
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:301]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 20]
```diff
@@ -165,169 +165,169 @@ $ ls
...
+  (gumamit ng "git add <file>..." upang i-update kung ano ang iko-commit)
+  (gumamit ng "git checkout -- <file>..." upang iwaksi ang mga pagbabago sa tinatrabahuang direktoryo)
 
-	modified:   .gitmodules
-	modified:   DbConnector (new commits)
+	nabago:   .gitmodules
+	nabago:   DbConnector (mga bagong commit)
 
-Submodules changed but not updated:
+Mga submodule na nabago ngunit hindi na-update:
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:318]: submodule  to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 21]
```diff
@@ -165,169 +165,169 @@ $ ls
...
 
 * DbConnector c3f01dc...c87d55d (4):
-  > catch non-null terminated lines
+  > hulihin ang di-null na tinatapos na mga linya
 ----
 
-At this point if you run `git diff` we can see both that we have modified our `.gitmodules` file and also that there are a number of commits that we've pulled down and are ready to commit to our submodule project.
+Sa puntong ito kung patakbuhin mo ang `git diff` maaari nating parehong makita na binago natin ang ating` .gitmodules` na file at na mayroon ding isang bilang ng mga commit na ating na-pull pababa at handa na i-commit sa ating submodule na proyekto.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:324]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 22]
```diff
@@ -336,26 +336,26 @@ index 6fc0b3d..fd1cc29 100644
...
-  > better connection routine
+  > hulihin ang di-null na tinatapos na mga linya
+  > mas matapang na pag-aasikaso ng kamalian
+  > mas mahusay na kalakaran ng db
+  > mas mahusay na kalakaran ng koneksyon
 ----
 
-This is pretty cool as we can actually see the log of commits that we're about to commit to in our submodule.
-Once committed, you can see this information after the fact as well when you run `git log -p`.
+Ito ay medyo cool bilang maaari nating aktwal na makita ang log ng mga commit na ating iko-commit sa ating submodule.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc]: cool  to astig

Comment (mind-blast) [reply book/07-git-tools/sections/submodules.asc]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 23]
```diff
@@ -364,25 +364,25 @@ index 6fc0b3d..fd1cc29 100644
...
-  > more efficient db routine
-  > better connection routine
+  > hulihin ang di-null na tinatapos na mga linya
+  > mas matapang na pag-aasikaso ng kamalian
+  > mas mahusay na kalakaran ng db
+  > mas mahusay na kalakaran ng koneksyon
 ----
 
-Git will by default try to update *all* of your submodules when you run `git submodule update --remote` so if you have a lot of them, you may want to pass the name of just the submodule you want to try to update.
+Bilang default, susubukan ng Git na i-update ang *lahat* ng iyong mga submodule kapag nagpatakbo ka ng `git submodule update --remote` kaya kung mayroon kang maraming mga ito, maaari mong gustuhin na ipasa ang pangalan ng submodule lamang na gusto mong subukang i-update.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:373]: submodule  to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 24]
```diff
@@ -364,25 +364,25 @@ index 6fc0b3d..fd1cc29 100644
...
 
-===== Working on a Submodule
+===== Pagtatrabaho sa isang Submodule
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:375]: Submodule to Submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 25]
```diff
@@ -364,25 +364,25 @@ index 6fc0b3d..fd1cc29 100644
...
 
-It's quite likely that if you're using submodules, you're doing so because you really want to work on the code in the submodule at the same time as you're working on the code in the main project (or across several submodules).
-Otherwise you would probably instead be using a simpler dependency management system (such as Maven or Rubygems).
+Malamang na kung gumagamit ka ng mga submodule, ginagawa mo ito dahil gusto mo talagang magtrabaho sa code sa submodule sa parehong oras habang nagtatrabaho ka sa code sa pangunahing proyekto (o sa ilang mga submodule).
+Kung hindi man ay malamang na ikaw ay gumamit ng isang mas simple na sistema ng pamamahala ng dependensya (tulad ng Maven o Rubygems).
 
-So now let's go through an example of making changes to the submodule at the same time as the main project and committing and publishing those changes at the same time.
+Kaya ngayon ay talakayin natin ang isang halimbawa ng paggawa ng mga pagbabago sa submodule sa parehong oras sa pangunahing proyekto at pag-commit at paglalathala sa mga pagbabago iyon sa parehong oras.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:380]: submodule  to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 26]
```diff
@@ -364,25 +364,25 @@ index 6fc0b3d..fd1cc29 100644
...
 
-Git will by default try to update *all* of your submodules when you run `git submodule update --remote` so if you have a lot of them, you may want to pass the name of just the submodule you want to try to update.
+Bilang default, susubukan ng Git na i-update ang *lahat* ng iyong mga submodule kapag nagpatakbo ka ng `git submodule update --remote` kaya kung mayroon kang maraming mga ito, maaari mong gustuhin na ipasa ang pangalan ng submodule lamang na gusto mong subukang i-update.
 
-===== Working on a Submodule
+===== Pagtatrabaho sa isang Submodule
 
-It's quite likely that if you're using submodules, you're doing so because you really want to work on the code in the submodule at the same time as you're working on the code in the main project (or across several submodules).
-Otherwise you would probably instead be using a simpler dependency management system (such as Maven or Rubygems).
+Malamang na kung gumagamit ka ng mga submodule, ginagawa mo ito dahil gusto mo talagang magtrabaho sa code sa submodule sa parehong oras habang nagtatrabaho ka sa code sa pangunahing proyekto (o sa ilang mga submodule).
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:377]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 27]
```diff
@@ -364,25 +364,25 @@ index 6fc0b3d..fd1cc29 100644
...
+Kaya ngayon ay talakayin natin ang isang halimbawa ng paggawa ng mga pagbabago sa submodule sa parehong oras sa pangunahing proyekto at pag-commit at paglalathala sa mga pagbabago iyon sa parehong oras.
 
-So far, when we've run the `git submodule update` command to fetch changes from the submodule repositories, Git would get the changes and update the files in the subdirectory but will leave the sub-repository in what's called a ``detached HEAD'' state.
-This means that there is no local working branch (like ``master'', for example) tracking changes.
-With no working branch tracking changes, that means even if you commit changes to the submodule, those changes will quite possibly be lost the next time you run `git submodule update`.
-You have to do some extra steps if you want changes in a submodule to be tracked.
+Sa ngayon, kapag pinatakbo natin ang utos na `git submodule update` upang makuha ang mga pagbabago mula sa mga repositoryo ng submodule, makakakuha ang Git ng mga pagbabago at i-update ang mga file sa subdirectory ngunit iiwan ang sub-repository sa tinatawag na `estado na `nakahiwalay na HEAD''.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:382]: subdirectory  to subdirektoryo

File: book/07-git-tools/sections/submodules.asc [hunk 28]
```diff
@@ -364,25 +364,25 @@ index 6fc0b3d..fd1cc29 100644
...
+Ito ay nangangahulugan na walang lokal na tinatrabahuang branch (tulad ng ``master'', halimbawa) na nagsusubaybay ng mga pagbabago.
+Ng walang tinatrabahuang branch na nagsusubaybay ng mga pagbabago, iyon ay nangangahulugan na kahit na ikaw ay mag-commit ng mga pagbabago sa submodule, ang mga pagbabagong iyon ay posibleng mawala sa susunod na patakbuhin mo ang `git submodule update`.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:384]: submodule to submodyul

File: book/07-git-tools/sections/submodules.asc [hunk 29]
```diff
@@ -364,25 +364,25 @@ index 6fc0b3d..fd1cc29 100644
...
+Kailangan mong gawin ang ilang mga dagdag na hakbang kung nais mong baguhin ang mga submodule na susubaybayan.
```
Comment (mind-blast) [inline book/07-git-tools/sections/submodules.asc:385]: submodule  to submodyul

**Comment by mind-blast** [comment]
Please find time to check my suggested words.

**Comment by heywens** [comment]
Hi Sir @robin-ho, thanks for the review. Some suggestions were applied except for the following words:
* submodule
* subdirectory

**Pull request merged**